### PR TITLE
[Spark Core][MINOR] fix "default partitioner cannot partition array keys" error message in PairRDDfunctions

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

In order to avoid confusing user,
error message in `PairRDDfunctions` 
`Default partitioner cannot partition array keys.`
is updated,
the one in `partitionBy` is replaced with
`Specified partitioner cannot partition array keys.`
other is replaced with
`Specified or default partitioner cannot partition array keys.`
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by srowen** [comment]
There are 5 instances of this check in the file -- they should all be handled the same way. I'm not sure this is accurate either because some code paths lead to these methods when HashPartitioner is used as a default. Just say that HashPartitioner can't be used? refactor one check method for this?

**Comment by WeichenXu123** [comment]
oh, there are 5 similar messages..
I check the others, the others may be set the default one,
so I update their message as "Specified or default partitioner..."
but the one in `partitionBy` must be set by user, can't use default one,
because the `partitionBy` API is let user specify how the RDD to be partitioned.
other cases, the `partitioner` parameter is optional and if user not specify it,
it will be set into the default `HashPartitioner`.
Now I update the code.
thanks!

**Comment by srowen** [comment]
Why bother saying 'specified' or 'default' at all though? it's probably even more informative to state that HashPartitioner doesn't work, no matter what the source. If the user specified HashPartitioner, that's clear. If they didn't, they'll still recognize that the other half of the message is relevant: some thing doesn't like their array keys.

File: core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala [hunk 1]
```diff
@@ -530,7 +530,7 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
    */
   def partitionBy(partitioner: Partitioner): RDD[(K, V)] = self.withScope {
     if (keyClass.isArray && partitioner.isInstanceOf[HashPartitioner]) {
-      throw new SparkException("Default partitioner cannot partition array keys.")
+      throw new SparkException("Specified partitioner cannot partition array keys.")
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala]: Even this method is called by other Spark code with HashPartitioner, which might lead to this error telling the user the "specified" partitioner doesn't work when the user code didn't specify a partitioner.

**Comment by WeichenXu123** [comment]
jenkins test please

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

**Comment by srowen** [comment]
OK, can't hurt to be clear and specific about this. Merged to master

**Pull request closed without merging**